plugins/synaptics-cape/fu-synaptics-cape-firmware.c

```diff
@@ -0,0 +1,215 @@
+/* -*- Mode: C; tab-width: 8; indent-tabs-mode: t; c-basic-offset: 8 -*-
+ *
+ * Copyright (C) 2021 Synaptics Incorporated <simon.ho@synaptics.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include <fwupdplugin.h>
+
+#include <string.h>
+
+#include "fu-synaptics-cape-firmware.h"
+
+typedef struct __attribute__((packed)) {
+	guint32 data[8];
+} FuCapeHidFwCmdUpdateWritePar;
+
+struct _FuSynapticsCapeFirmware {
+	FuSrecFirmwareClass parent_instance;
+	gchar *version;
+	guint16 version_w;
+	guint16 version_x;
+	guint16 version_y;
+	guint16 version_z;
```

Comment:
these feel like they can just be local variables in the function while they are parsed, I don't see a need to make them part of the object

Reply:
Okay, will move them to local.